float32 position
float32 speed_fraction